### PR TITLE
refactor(transformer): shorten code in TS namespace transform

### PR DESCRIPTION
**Pull request by overlookmotel**
Refactor to shorten code.

### CODE REVIEW

**Comment by overlookmotel** [comment]
* **#3468** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3468?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_transformer/src/typescript/namespace.rs [hunk 1]
```diff
@@ -200,107 +197,85 @@ impl<'a> TypeScript<'a> {
...
-                Statement::VariableDeclaration(decl) => {
+                Statement::VariableDeclaration(ref decl) => {
                     is_empty = false;
                     decl.bound_names(&mut |id| {
                         names.insert(id.name.clone());
                     });
-                    new_stmts.push(Statement::VariableDeclaration(decl));
+                    new_stmts.push(stmt);
+                }
+                Statement::ExportNamedDeclaration(ref export_decl)
```
Comment (overlookmotel) [inline crates/oxc_transformer/src/typescript/namespace.rs:226]: @Dunqing This PR doesn't change the logic, but this doesn't look right to me. We can end up with `export {x}` in the output. [Playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAICsgICAgICAgIC3mMl9fWP2DOEtS29bAAVF3HDuYFci2KI8dKJg9t1AO3n4jQYRoCBjAGTVEvqeRiV%2FeJWQgA%3D%3D)

I'm not sure what the correct behaviour is, though.

File: crates/oxc_transformer/src/typescript/namespace.rs [hunk 2]
```diff
@@ -200,107 +197,85 @@ impl<'a> TypeScript<'a> {
...
+                    if export_decl.declaration.is_none() =>
+                {
+                    // TODO: This isn't correct. Need to transform `export {x}` somehow.
+                    new_stmts.push(stmt);
```
Comment (overlookmotel) [inline crates/oxc_transformer/src/typescript/namespace.rs:230]: @Dunqing This PR doesn't change the logic, but this doesn't look right to me. We can end up with `export {x}` in the output. [Playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAICsgICAgICAgIC3mMl9fWP2DOEtS29bAAVF3HDuYFci2KI8dKJg9t1AO3n4jQYRoCBjAGTVEvqeRiV%2FeJWQgA%3D%3D)

I'm not sure what the correct behaviour is, though.

**Comment by overlookmotel** [comment]
This PR is pure refactor. Have only requested your review @Dunqing on the one point highlighted above.

File: crates/oxc_transformer/src/typescript/namespace.rs (hunk 2 above)
Comment (Dunqing) [reply crates/oxc_transformer/src/typescript/namespace.rs:230]: You are right. We don't worry about it. This is not valid code in typescript https://www.typescriptlang.org/play/?verbatimModuleSyntax=true#code/BQMwrgdgxgLglgewgKmASgAQG8BQH8bjTxLAAeGAvBgJ5wCmANgCYYDMmWAvjl2jkA

We can also add a check for `export {x}` inside the namespace in https://github.com/oxc-project/oxc/blob/05-29-refactor_typescript_namespace_reuse_tsmoduleblock_s_scope_id/crates/oxc_semantic/src/checker/typescript.rs#L169

File: crates/oxc_transformer/src/typescript/namespace.rs [hunk 3]
```diff
@@ -200,107 +197,85 @@ impl<'a> TypeScript<'a> {
...
                     decl.bound_names(&mut |id| {
                         names.insert(id.name.clone());
                     });
-                    new_stmts.push(Statement::VariableDeclaration(decl));
+                    new_stmts.push(stmt);
+                }
+                Statement::ExportNamedDeclaration(ref export_decl)
+                    if export_decl.declaration.is_none() =>
+                {
+                    // TODO: This isn't correct. Need to transform `export {x}` somehow.
```
Comment (Dunqing) [inline crates/oxc_transformer/src/typescript/namespace.rs:229]:
```suggestion
```

**Review by Dunqing — APPROVED** [review]
I very like this improvement!

File: crates/oxc_transformer/src/typescript/namespace.rs (hunk 2 above)
Comment (overlookmotel) [reply crates/oxc_transformer/src/typescript/namespace.rs:230]: So can we just remove this branch entirely? Do nothing rather than `new_stmts.push(stmt)`?

Comment (Dunqing) [reply crates/oxc_transformer/src/typescript/namespace.rs:230]: Yes, we can.

Comment (overlookmotel) [reply crates/oxc_transformer/src/typescript/namespace.rs:230]: Thanks for coming back. I'll make that change before we merge this.

Comment (overlookmotel) [reply crates/oxc_transformer/src/typescript/namespace.rs]: That branch removed in #3475.

**Comment by overlookmotel** [comment]
@Dunqing I pushed another commit removing `is_empty` to remove a few more lines. If you're happy with it, please merge. I'll get onto the substantive changes to fix the scopes tomorrow.

**Review by Dunqing — APPROVED** [review]
LGTM

**Pull request merged**